src/coreclr/jit/lowerxarch.cpp

```diff
@@ -7503,6 +7501,50 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
             return false;
         }
 
+        case NI_SSE3_MoveAndDuplicate:
+        case NI_AVX2_BroadcastScalarToVector128:
+        case NI_AVX2_BroadcastScalarToVector256:
+        case NI_AVX512F_BroadcastScalarToVector512:
+        {
+            // make the broadcast node containable when embedded broadcast can be enabled.
+            if (intrinsicId == NI_SSE3_MoveAndDuplicate)
+            {
+                // NI_SSE3_MoveAndDuplicate is for Vector128<double> only.
+                assert(childNode->AsHWIntrinsic()->GetSimdBaseType() == TYP_DOUBLE);
+            }
+            if (comp->compOpportunisticallyDependsOn(InstructionSet_AVX512F_VL) &&
+                parentNode->OperIsEmbBroadcastCompatible())
+            {
+                GenTree* broadcastOperand = childNode->AsHWIntrinsic()->Op(1);
+                bool     childSupportsRegOptional;
+                if (broadcastOperand->OperIs(GT_LCL_VAR))
+                {
+                    const unsigned opLclNum = broadcastOperand->AsLclVar()->GetLclNum();
+                    comp->lvaSetVarDoNotEnregister(opLclNum DEBUGARG(DoNotEnregisterReason::LiveInOutOfHandler));
+                }
+                else if (broadcastOperand->OperIs(GT_HWINTRINSIC) &&
+                         broadcastOperand->AsHWIntrinsic()->Op(1)->OperIs(GT_LCL_VAR))
+                {
+                    assert(broadcastOperand->AsHWIntrinsic()->GetHWIntrinsicId() == NI_Vector128_CreateScalarUnsafe);
+                    const unsigned opLclNum = broadcastOperand->AsHWIntrinsic()->Op(1)->AsLclVar()->GetLclNum();
+                    comp->lvaSetVarDoNotEnregister(opLclNum DEBUGARG(DoNotEnregisterReason::LiveInOutOfHandler));
+                }
+                if (IsContainableHWIntrinsicOp(childNode->AsHWIntrinsic(), broadcastOperand, &childSupportsRegOptional))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        case NI_AVX_BroadcastScalarToVector128:
+        case NI_AVX_BroadcastScalarToVector256:
+        {
+            GenTree* childNodeOp = hwintrinsic->Op(1);
+            assert(childNodeOp->OperIs(GT_LCL_ADDR, GT_LCL_VAR));
```

Comment:
I think this assert is going to fire for a number of real world cases; including `LCL_FLD`, `IND`, etc

We should always be able to contain the `AVX_BroadcastScalarToVector` case. Its child node may or may not be contained (it can in the same cases that `GT_IND`'s child node can be contained), but that just makes a difference if we emit `[reg]` (the address is pre-evaluated) or if we're allowed to emit something like `[reg1+reg2*scale+disp]` (that is a complex addressing mode)